src/rodrigues_params.jl

```diff
@@ -0,0 +1,131 @@
+
+"""
+    RodriguesParam{T}
+
+Rodrigues parameters are a three-dimensional parameterization of rotations.
+They have a singularity at 180° but do not inherit the sign ambiguities of quaternions
+or MRPs
+"""
+struct RodriguesParam{T} <: Rotation{3,T}
+    x::T
+    y::T
+    z::T
+    @inline function RodriguesParam{T}(x, y, z) where T
+        new{T}(x, y, z)
+    end
+end
+
+# ~~~~~~~~~~~~~~~ Constructors ~~~~~~~~~~~~~~~ #
+RodriguesParam(x::X, y::Y, z::Z) where {X,Y,Z} = RodriguesParam{promote_type(X,Y,Z)}(x, y, z)
+(::Type{RP})(g::StaticVector) where RP<:RodriguesParam = RP(g[1], g[2], g[3])
+
+# ~~~~~~~~~~~~~~~ Conversions ~~~~~~~~~~~~~~~ #
+SVector(g::RodriguesParam) = SVector{3}(g.x, g.y, g.z)
```

Comment:
Are you sure about this one? This is somewhat equivalent to defining a method like `SVector{3}(::SMatrix{3,3})` - the `AbstractArray` and `StaticArray` interfaces have to take precedence over the special orthogonal properties of the rotation matrices and things we'd like to do with our parameterizations of them, so I'd be tempted to make this a seperate function.

Reply:
Yeah, we can make it a separate function. What about `params` or `rotation_params`? This function could be nice as an easy way to check the number of parameters a rotation representation uses.